compiler/src/dmd/aggregate.d

```diff
@@ -829,7 +830,11 @@ public uint placeField(ref uint nextoffset, uint memsize, uint memalignsize,
     bool overflow;
     const sz = addu(memsize, actualAlignment, overflow);
     addu(ofs, sz, overflow);
-    if (overflow) assert(0);
+    if (overflow)
+    {
+        error(loc, "max object size %u exceeded from adding %u + %u + %u when placing field in aggregate", uint.max, memsize, actualAlignment, ofs);
```

Comment:
It'd be nice to explain the %u+%u+%u to the user (field size + alignment adjustment + field offset). Field name would be nice too...

Reply:
fixed